test/integration/containerd_test.go

```diff
@@ -0,0 +1,136 @@
+// +build integration
+
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/docker/machine/libmachine/state"
+	"k8s.io/minikube/test/integration/util"
+)
+
+func TestContainerd(t *testing.T) {
+	p := profile(t)
+	if isTestNoneDriver() {
+		t.Skip("Can't run containerd backend with none driver")
+	}
+	t.Parallel()
+	mk := NewMinikubeRunner(t, p)
+	if !isTestNoneDriver() { // none driver doesn't need to be deleted
+		defer mk.TearDown(t)
+	}
+
+	stdout, stderr, err := mk.Start("--container-runtime=containerd", "--docker-opt containerd=/var/run/containerd/containerd.sock")
+	if err != nil {
+		t.Fatalf("failed to start minikube (for profile %s) failed : %v\nstdout: %s\nstderr: %s", p, err, stdout, stderr)
+	}
+
+	t.Run("group", func(t *testing.T) {
+		t.Run("Gvisor", testGvisor)
+		t.Run("GvisorRestart", testGvisorRestart)
+	})
+}
+
+func testGvisor(t *testing.T) {
```

Comment:
Can this go into a gvisor_test.go file?

Reply:
only containerd uses it ! and also I was thinking the not main functions should not be in a _test.go filename since golang only looks at _test.go files and the others get called by main tests...wdyt ?